src/transformState.js

```diff
@@ -4,8 +4,41 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+const removeKey = function(actionObject, object) {
+  const { keysToRemove } = actionObject;
+
+  for (const keyToRemove of keysToRemove) {
+    delete object[keyToRemove];
+  }
+};
+const clearProperties = function(actionObject, object) {
+  for (const key in object) {
+    if (object.hasOwnProperty(key)) {
+      delete object[key];
+    }
+  }
```

Comment:
you are already iterating through the keys of the object, do you really need to check whether they have the property?

Also, you can just reset the object itself instead of removing properties one by one

Reply:
> Also, you can just reset the object itself instead of removing properties one by one

do you mean stateCopy = {} ? but I'll lose the link to the state object, cause i should make changes in the state. also, i can't make assignments to a function parameter